src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -421,7 +421,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !node.FullSpan.IntersectsWith(node.FullSpan))
+            if (node == null || !this.FullSpan.IntersectsWith(node.FullSpan))
```

Comment:
I think so. 
I was wondering about this too, and the only case where Span comparison wouldn't work is if the nodes aren't in the same tree. I didn't realize there is an easy way to check that.
I'll make the change.